lib/src/commit_builder.rs

```diff
@@ -109,11 +109,18 @@ impl CommitBuilder<'_> {
         self.commit.root_tree.as_legacy_tree_id()
     }
 
+    // TODO(#1624): delete when all callers have been updated to support tree-level
+    // conflicts
     pub fn set_tree(mut self, tree_id: TreeId) -> Self {
```

Comment:
You could rename it to `set_legacy_tree_id`.

I found the comment mildly confusing since you can, for now, use `MergedTreeId` with legacy trees (and that's what this PR does).

Feel free to keep everything as is if this is not going to be the case for long.

Reply:
Yes, I hope to delete this pretty soon, so I'll leave it as is.